support/scripts/install

```diff
@@ -3,67 +3,219 @@
+    echo "                              your OS automatically"
+    echo
+}
+
+
+validate_OS_type() {
+
+    if [[ -z "${OS}" ]]; then
+        echo "ERROR: Unable to determine your system type."
         exit 1
     fi
+}
+
+
+validate_supported_OS() {
+
+    SUPPORTED_OS_TYPES=(
+        "Linux_x86_64"
+        "Linux_armv6"
+        "Linux_arm64"
+        "Linux_i386"
+        "Darwin_x86_64"
+        "Darwin_arm64"
+    )
+
+    if [[ "${SUPPORTED_OS_TYPES[*]}" =~ "${OS}" ]]; then
+        return 0
+    else
+        echo "Error: $OS is not supported, installation will attempt to proceed manually."
+        echo
+        echo "Please check the releases page for a list of supported systems."
+        echo "https://github.com/sundowndev/phoneinfoga/releases"
+        echo
+        echo "Read more at https://sundowndev.github.io/phoneinfoga/install/"
+        echo
+        AUTOMATIC=1
+        return 1
+    fi
+}
+
+
+validate_checksum() {
+
+    local WGET_INSTALLED=false
+    which wget > /dev/null
+    [ $? -eq 1 ] || WGET_INSTALLED=true
+
+    echo "Validating checksum ..."
+    if $WGET_INSTALLED; then
+        wget --quiet --show-progress "https://github.com/sundowndev/phoneinfoga/releases/download/$PHONEINFOGA_VERSION/phoneinfoga_checksums.txt"
+    elif ! $WGET_INSTALLED; then
+        curl --progress-bar -LOC - "https://github.com/sundowndev/phoneinfoga/releases/download/$PHONEINFOGA_VERSION/phoneinfoga_checksums.txt"
+    fi
+
+    local SHA256SUM_INSTALLED=false
+    which sha256sum > /dev/null
+    [ $? -eq 1 ] || SHA256SUM_INSTALLED=true
+
+    if $SHA256SUM_INSTALLED; then
+        sha256sum --ignore-missing -c phoneinfoga_checksums.txt
+        [ $? -eq 0 ] || exit 1
+    elif ! $SHA256SUM_INSTALLED; then
+        shasum --ignore-missing -c phoneinfoga_checksums.txt
+        [ $? -eq 0 ] || exit 1
+    fi
+
+    rm phoneinfoga_checksums.txt
+}
+
+
+download_latest_release() {
+
+    local WGET_INSTALLED=false
+    which wget > /dev/null
+    [ $? -eq 1 ] || local WGET_INSTALLED=true
+
+    echo "https://github.com/sundowndev/phoneinfoga/releases/download/$PHONEINFOGA_VERSION/phoneinfoga_$OS.tar.gz"
+
+    if $WGET_INSTALLED; then
+        wget --quiet --show-progress "https://github.com/sundowndev/phoneinfoga/releases/download/$PHONEINFOGA_VERSION/phoneinfoga_$OS.tar.gz"
+    elif ! $WGET_INSTALLED; then
+        curl --progress-bar -LOC - "https://github.com/sundowndev/phoneinfoga/releases/download/$PHONEINFOGA_VERSION/phoneinfoga_$OS.tar.gz"
+    fi
+}
+
+
+manual_download_latest_release() {
 
-else
-    sha256sum --ignore-missing -c phoneinfoga_SHA256SUMS
+    ASSETS=$(curl -s https://api.github.com/repos/sundowndev/phoneinfoga/releases/latest | jq --raw-output '.assets[] | .name' | grep --color=never tar.gz)
+
+    local WGET_INSTALLED=false
+    which wget > /dev/null
+    [ $? -eq 1 ] || WGET_INSTALLED=true
+
+    echo "Please select the version of PhoneInfoga that you need."
+    PS3="Enter a number: "
```

Comment:
What system are you using? It was working on my Mac. I'll do some more testing on my Arch machine before updating

Reply:
When you get the chance, can you please give it another try? I think these updates should resolve this bug.

Reply:
Still doesn't work on my end, I think you should use `read` instead

Reply:
Sorry one more question, because I'm struggling to find an explanation for the behavior you're seeing. Did you run this code block separately from the rest of the script? I see `curl -sSL https://...` but that is not part of the original code.

Otherwise, can you please explain what you mean by using `read`? Do you mean instead of `select`? Thanks!

Reply:
I was able to replicate the issue, it is an oddity of `bash` and I was able to get around it using this command to fetch & run the script instead.

#### Suggested solution:
Update the documentation to include this command instead.
```
bash <( curl -sSL https://raw.githubusercontent.com/sundowndev/phoneinfoga/78c4478f7b6988531e041e9196b0ca768cef975e/support/scripts/install )
```
<br>

this solution also works perfectly fine in automatic mode (above example) or when adding single/multiple arguments to the end of the command, like this.
```
bash <( curl -sSL https://raw.githubusercontent.com/sundowndev/phoneinfoga/78c4478f7b6988531e041e9196b0ca768cef975e/support/scripts/install ) -m -s
```

<br>

[This](https://stackoverflow.com/a/4642978) is where I discovered the solution, after a few failed other attempts.

Updating the one-liner in your documentation's [Getting Started](https://sundowndev.github.io/phoneinfoga/getting-started/install/#binary-installation-recommended) should remedy this behavior.

Reply:
Fair enough, thanks for the research!